dlib/math/constants.h

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2006  Davis E. King (davis@dlib.net)
+// License: Boost Software License   See LICENSE.txt for the full license.
+#ifndef DLIB_MATH_CONSTANTS
+#define DLIB_MATH_CONSTANTS
+
+namespace dlib
+{
+    template<typename R>
+    struct numeric_constants
+    {
+        static R pi() noexcept
```

Comment:
There is already dlib/numeric_constants.h

Reply:
Ok latest commit addresses this.